ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -97,6 +97,17 @@ extension BrowserViewController: WKNavigationDelegate {
         hideReaderModeBar(animated: false)
       }
     }
+
+    if let tab = tab(for: webView) {
+      tab.redirectURLs = []
+
+      // Add current WebView URL to redirect chain as the source URL. The redirect
+      // chain must be sorted by the order of redirects with the first URL being
+      // the source URL.
+      if let url = webView.url, rewards.isEnabled {
+        tab.redirectURLs.append(url)
+      }
+    }
```

Comment:
Can we please decouple the code as we use similar logic in two places, thanks

Reply:
This is WKNavigationDelegate which makes sense the code is here, but we should remove code duplication

Reply:
But it would be great if redirectChain is fixed for any callers, the current fix is only for ads.